lib/ConstExtract/ConstExtract.cpp

```diff
@@ -128,69 +128,116 @@ static std::string extractLiteralOutput(Expr *expr) {
   return LiteralOutput;
 }
 
-static std::shared_ptr<CompileTimeValue>
-extractPropertyInitializationValue(VarDecl *propertyDecl) {
-  auto binding = propertyDecl->getParentPatternBinding();
-  if (binding) {
-    auto originalInit = binding->getOriginalInit(0);
-    if (originalInit) {
-      auto literalOutput = extractLiteralOutput(originalInit);
-      if (!literalOutput.empty()) {
-        return std::make_shared<RawLiteralValue>(literalOutput);
-      }
+static std::shared_ptr<CompileTimeValue> extractCompileTimeValue(Expr *expr) {
+  switch (expr->getKind()) {
+  case ExprKind::Array:
+  case ExprKind::Dictionary:
```

Comment:
@quinntaylor - One reason I restructured this was to help us extract out `Array` and `Dictionary` into their own `CompileTimeValue` next.